libraries/engage/product/components/UnitQuantityPicker/UnitQuantityPicker.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { css } from 'glamor';
+import classNames from 'classnames';
+import { themeConfig } from '@shopgate/engage';
+import { RippleButton, QuantityInput } from '@shopgate/engage/components';
+import { useWidgetSettings } from '@shopgate/engage/core';
+
+const { variables, colors } = themeConfig;
+
+const styles = {
+  root: css({
+    display: 'flex',
+    flexDirection: 'row',
+  }).toString(),
+  input: (color, bgColor) => css({
+    padding: `0 ${variables.gap.small}px`,
+    textAlign: 'center',
+    flex: 1,
+    fontSize: 15,
+    height: 28,
+    width: '100%',
+    backgroundColor: bgColor,
+    color,
+  }).toString(),
+  inputWrapper: css({
+    width: '100%',
+  }),
+  button: (color, bgColor) => css({
+    backgroundColor: `${bgColor} !important`,
+    color: `${color} !important`,
+    width: 28,
+    ' &&': {
+      minWidth: 28,
+      padding: 0,
+    },
+    height: 28,
+  }).toString(),
+  buttonRipple: css({
+    padding: 0,
+  }).toString(),
+  buttonNoRadiusLeft: css({
+    ' &&': {
+      borderTopLeftRadius: 0,
+      borderBottomLeftRadius: 0,
+    },
+  }).toString(),
+  buttonNoRadiusRight: css({
+    ' &&': {
+      borderTopRightRadius: 0,
+      borderBottomRightRadius: 0,
+    },
+  }).toString(),
+  disabled: css({
+    ' > div': {
+      padding: 0,
+    },
+  }).toString(),
+};
+
+/**
+ * A Quantity Picker with unit support.
+ * @returns {JSX}
+ */
+const UnitQuantityPicker = ({
+  className,
+  onChange,
+  value,
+  allowDecrement,
+  allowIncrement,
+  allowZero,
+  decrementStep,
+  incrementStep,
+  maxDecimals,
+  unit,
+  disabled,
+  minValue,
+  maxValue,
+}) => {
+  const {
```

Comment:
defaults here are different than defaults in extension config.
i will update the extension config for you